### PR TITLE
Generate github release executables for macos

### PR DESCRIPTION
**Pull request by nitinprakash96**
Changes include:
- Generating executables for macos (x64 and arm)
- Simplifications to GHC and cabal setup
- Bump GH release action
- Use `softprops/action-gh-release@v1` for uploading release assets instead of archived `actions/upload-release-asset@v1.0.1`

Example release set: https://github.com/nitinprakash96/swarm-copy/releases/tag/0.8.2.8
And workflow: https://github.com/nitinprakash96/swarm-copy/actions/runs/10808335665

### CODE REVIEW

**Comment by byorgey** [comment]
This certainly looks much simpler!  But how is it possible to test actions like this?

**Comment by xsebek** [comment]
Maybe we could run this in a repo copy with a commented-out upload to Hackage?

**Review by xsebek — APPROVED** [review]
Looks good; I can download and run both on my M2 MacBook:
```bash
curl -LO https://github.com/nitinprakash96/swarm-copy/releases/download/0.8.2.8/swarm-Darwin-x86_64
chmod +x ./swarm-Darwin-x86_64
xattr -d com.apple.quarantine swarm-Darwin-x86_64  # removes "malware not checked" popup
./swarm-Darwin-x86_64
```
Please check the data ZIP. The other things are just nitpicks. 🙂

File: .github/workflows/haskell-release.yml [hunk 1]
```diff
@@ -10,178 +10,135 @@ on:
...
+          with:
+            name: swarm
+            draft: false
+            prerelease: false
+            files: |
+              swarm-Linux-x86_64
+              swarm-Darwin-x86_64
+              swarm-Darwin-arm64
+              LICENSE
+              swarm-data.zip
```
[the remark this answers is not in the document]
Comment (nitinprakash96) [reply .github/workflows/haskell-release.yml:139]: Good catch. This is a miss on my part. Here's the updated version:
Release - https://github.com/nitinprakash96/swarm-copy/releases/tag/0.8.3.2
And build - https://github.com/nitinprakash96/swarm-copy/actions/runs/10824383879

**Comment by byorgey** [comment]
@nitinprakash96 So is this ready to merge?

**Comment by nitinprakash96** [comment]
@byorgey If there are no more reviews then I'd say we can merge this.

**Comment by byorgey** [comment]
I know I said I would look at this but @xsebek knows much more than me about the release process so I don't think I would have anything to add.  Let's merge it!

**Pull request merged**